Docs for channel upgrades CLI (#5553)
cosmos/ibc-go: docs/docs/01-ibc/06-channel-upgrades.md

> IBC applications should not attempt to process any packet data under the new conditions until after `OnChanUpgradeOpen`
> has been executed, as up until this point it is still possible for the upgrade handshake to fail and for the channel
> to remain in the pre-upgraded state. 

## Upgrading channels with the CLI

A new cli has been added which enables either

>     - submitting a governance proposal which contains a `MsgChanUpgradeInit` for every channel to be upgraded.

Reviewer: ```suggestion
    - submitting a governance proposal which contains a `MsgChannelUpgradeInit` for every channel to be upgraded.
```

>     - generation a `proposal.json` file which contains the proposal contents to be edited/submitted at a later date.

Reviewer 2: ```suggestion
    - generating a `proposal.json` file which contains the proposal contents to be edited/submitted at a later date.
```

> The following example, would submit a governance proposal with the specified deposit, title and summary which would
> contain a `MsgChanUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.

Reviewer 2: ```suggestion
The following example, would submit a governance proposal with the specified deposit, title and summary while also containing a `MsgChannelUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```

Reviewer 3: i kind of prefer @chatton's version tbh bc the proposal is there to contain (approve) the `MsgChannelUpgradeInit`, but it's a super nit really.

Author: will leave as is if you guys are okay with it then?

Reviewer 2: ah, maybe just the `MsgChannelUpgradeInit` as opposed to `MsgChanUpgradeInit`? (git diff isn't particularly good at showing the specific diffs).

Wording is perfectly fine by me either way!

Author: ah missed the typo, thanks @DimitrisJim

> contain a `MsgChanUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.

Reviewer: ```suggestion
contain a `MsgChannelUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```

> Note: by adding the `--json` flag, the command would instead output the contents of the proposal which could be 
> stored in a `proposal.json` file to be edited and submitted at a later date.

```bash
simd tx ibc channel upgrade-channels "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}" \
  --deposit "10stake" \
  --title "Channel Upgrades Governance Proposal" \
  --summary "Upgrade all transfer channels to be fee enabled" \
  --port-pattern "transfer"
```

It is also possible to explicitly list a comma separated string of channel IDs. It is important to note that the 
regular expression matching specified by `--port-pattern` (which defaults to `transfer`) still applies.

For example the following command would generate the contents of a `proposal.json` file which would attempt to upgrade

> channels with a port ID of `transfer` and channelID of `channel-0`, `channel-1` or `channel-2`.

Reviewer 2: ```suggestion
channels with a port ID of `transfer` and a channelID of `channel-0`, `channel-1` or `channel-2`.
```
